File: CS/BackgroundGeolocation.Module.Web/Controllers/DisplayGeolocationController.cs
using System.Collections.Generic;
using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Maps.Web;
using DevExpress.ExpressApp.Maps.Web.Helpers;
using DevExpress.ExpressApp.SystemModule;
using DevExpress.Persistent.Base;
using BackgroundGeolocation.Module.BusinessObjects;
using Newtonsoft.Json;

namespace MobileTestApplication.Module.Web.Controllers {
    public class DisplayGeolocationController : ObjectViewController<ListView, DeviceLocation> {
        protected override void OnActivated() {
            base.OnActivated();

            ResetViewSettingsController resetViewSettingsController = Frame.GetController<ResetViewSettingsController>();
            if(resetViewSettingsController != null) {
                resetViewSettingsController.ResetViewSettingsAction.Active["EnabledInNestedMaps"] = false;
            }
        }
        protected override void OnDeactivated() {
            ResetViewSettingsController resetViewSettingsController = Frame.GetController<ResetViewSettingsController>();
            if(resetViewSettingsController != null) {
                resetViewSettingsController.ResetViewSettingsAction.Active["EnabledInNestedMaps"] = true;
            }

            base.OnDeactivated();
        }
        protected override void OnViewControlsCreated() {
            base.OnViewControlsCreated();

            WebMapsListEditor webMapsListEditor = View.Editor as WebMapsListEditor;
            if(webMapsListEditor != null)
                webMapsListEditor.MapViewer.ClientSideEvents.Customize = GetCustomizeScript();
        }
        private IEnumerable<MapPoint> GetPolylinePoints() {
            List<MapPoint> polylinePoints = new List<MapPoint>();
            foreach(IMapsMarker marker in View.CollectionSource.List) {
                polylinePoints.Add(new MapPoint(marker.Latitude, marker.Longitude));
            }
            return polylinePoints;
        }
        private string GetCustomizeScript() {
            var polylines = JsonConvert.SerializeObject(GetPolylinePoints());
            return string.Format(
    @"function(sender, map) {{
    map.on('ready', function(e) {{
        var googleMap = e.originalMap;
        var clientPathCoordinates = {0};
        
        var iconSettings = {{
            path: google.maps.SymbolPath.FORWARD_CLOSED_ARROW
        }};

        var clientPath = new google.maps.Polyline({{
            path: clientPathCoordinates,
            strokeColor: '#FF0000',
            strokeOpacity: 1.0,
            strokeWeight: 2,
            icons: [{{
                icon: iconSettings,
                repeat: '35px',
                offset: '100%'}}]
        }});

        window.clientPath = clientPath;
        clientPath.setMap(googleMap);
    }});
}}", polylines);
        }
    }
}

File: VB/BackgroundGeolocation.Module.Web/Controllers/DisplayGeolocationController.vb
Imports System.Collections.Generic
Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Maps.Web
Imports DevExpress.ExpressApp.Maps.Web.Helpers
Imports DevExpress.ExpressApp.SystemModule
Imports DevExpress.Persistent.Base
Imports BackgroundGeolocation.Module.BusinessObjects
Imports Newtonsoft.Json

Namespace MobileTestApplication.Module.Web.Controllers
	Public Class DisplayGeolocationController
		Inherits ObjectViewController(Of ListView, DeviceLocation)

		Protected Overrides Sub OnActivated()
			MyBase.OnActivated()

			Dim resetViewSettingsController As ResetViewSettingsController = Frame.GetController(Of ResetViewSettingsController)()
			If resetViewSettingsController IsNot Nothing Then
				resetViewSettingsController.ResetViewSettingsAction.Active("EnabledInNestedMaps") = False
			End If
		End Sub
		Protected Overrides Sub OnDeactivated()
			Dim resetViewSettingsController As ResetViewSettingsController = Frame.GetController(Of ResetViewSettingsController)()
			If resetViewSettingsController IsNot Nothing Then
				resetViewSettingsController.ResetViewSettingsAction.Active("EnabledInNestedMaps") = True
			End If

			MyBase.OnDeactivated()
		End Sub
		Protected Overrides Sub OnViewControlsCreated()
			MyBase.OnViewControlsCreated()

			Dim webMapsListEditor As WebMapsListEditor = TryCast(View.Editor, WebMapsListEditor)
			If webMapsListEditor IsNot Nothing Then
				webMapsListEditor.MapViewer.ClientSideEvents.Customize = GetCustomizeScript()
			End If
		End Sub
		Private Function GetPolylinePoints() As IEnumerable(Of MapPoint)
			Dim polylinePoints As New List(Of MapPoint)()
			For Each marker As IMapsMarker In View.CollectionSource.List
				polylinePoints.Add(New MapPoint(marker.Latitude, marker.Longitude))
			Next marker
			Return polylinePoints
		End Function
		Private Function GetCustomizeScript() As String
			Dim polylines = JsonConvert.SerializeObject(GetPolylinePoints())
			Return String.Format("function(sender, map) {{" & ControlChars.CrLf & _
"    map.on('ready', function(e) {{" & ControlChars.CrLf & _
"        var googleMap = e.originalMap;" & ControlChars.CrLf & _
"        var clientPathCoordinates = {0};" & ControlChars.CrLf & _
"        " & ControlChars.CrLf & _
"        var iconSettings = {{" & ControlChars.CrLf & _
"            path: google.maps.SymbolPath.FORWARD_CLOSED_ARROW" & ControlChars.CrLf & _
"        }};" & ControlChars.CrLf & _
ControlChars.CrLf & _
"        var clientPath = new google.maps.Polyline({{" & ControlChars.CrLf & _
"            path: clientPathCoordinates," & ControlChars.CrLf & _
"            strokeColor: '#FF0000'," & ControlChars.CrLf & _
"            strokeOpacity: 1.0," & ControlChars.CrLf & _
"            strokeWeight: 2," & ControlChars.CrLf & _
"            icons: [{{" & ControlChars.CrLf & _
"                icon: iconSettings," & ControlChars.CrLf & _
"                repeat: '35px'," & ControlChars.CrLf & _
"                offset: '100%'}}]" & ControlChars.CrLf & _
"        }});" & ControlChars.CrLf & _
ControlChars.CrLf & _
"        window.clientPath = clientPath;" & ControlChars.CrLf & _
"        clientPath.setMap(googleMap);" & ControlChars.CrLf & _
"    }});" & ControlChars.CrLf & _
"}}", polylines)
		End Function
	End Class
End Namespace
